opm/simulators/linalg/bda/rocsparseSolverBackend.cpp

```diff
@@ -526,21 +537,10 @@ void rocsparseSolverBackend<block_size>::solve_system(WellContributions &wellCon
     Timer t;
 
     // actually solve
-    try {
-        gpu_pbicgstab(wellContribs, res);
-    } catch (const cl::Error& error) {
-        std::ostringstream oss;
-        oss << "rocsparseSolverBackend::solve_system error: " << error.what() << "(" << error.err() << ")\n";
-        oss << getErrorString(error.err());
-        // rethrow exception
-        OPM_THROW(std::logic_error, oss.str());
-    } catch (const std::logic_error& error) {
-        // rethrow exception by OPM_THROW in the try{}, without this, a segfault occurs
-        throw error;
-    }
+    gpu_pbicgstab(wellContribs, res);
+    HIP_CHECK(hipStreamSynchronize(stream));
```

Comment:
I want to learn. Why is the synchonization needed?
Previously, it was only done for a high verbosity level to make sure the reported times are correct.

Reply:
It was originally implemented to match the cusolverBackend.cu implementation. But yes this additional synchronization isn't necessary